src/data/polynomial/ring_division.lean

```diff
@@ -918,6 +918,13 @@ lemma card_roots_le_map [is_domain A] [is_domain B] {p : A[X]} {f : A →+* B} (
   p.roots.card ≤ (p.map f).roots.card :=
 by { rw ← p.roots.card_map f, exact multiset.card_le_of_le (map_roots_le h) }
 
+lemma card_roots_le_map_of_injective [is_domain A] [is_domain B] {p : A[X]} {f : A →+* B}
```

Comment:
Can you please add a version using `[no_zero_smul_divisors]`, so Lean can find it by itself?

Reply:
Hmm, I'm not sure what you mean, `polynomial.roots` requires `is_domain` currently, so that can't be generalized in this PR really. Are you talking about an algebra map version?

Reply:
Sorry, I somehow mixed up injectivity of `f` with injectivity of (a non existent here) `algebra_map`. Forget my comment.